static/app/components/events/featureFlags/setupIntegrationModal.tsx

```diff
@@ -0,0 +1,216 @@
+import {Fragment, useCallback, useState} from 'react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import {addSuccessMessage} from 'sentry/actionCreators/indicator';
+import type {ModalRenderProps} from 'sentry/actionCreators/modal';
+import Alert from 'sentry/components/alert';
+import {Button, LinkButton} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import type {Data} from 'sentry/components/forms/types';
+import TextCopyInput from 'sentry/components/textCopyInput';
+import {IconWarning} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import OrganizationStore from 'sentry/stores/organizationStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import {space} from 'sentry/styles/space';
+import {defined} from 'sentry/utils';
+import useApi from 'sentry/utils/useApi';
+
+export type ChildrenProps<T> = {
+  Body: (props: {children: React.ReactNode}) => ReturnType<ModalRenderProps['Body']>;
+  Footer: () => ReturnType<ModalRenderProps['Footer']>;
+  Header: (props: {children: React.ReactNode}) => ReturnType<ModalRenderProps['Header']>;
+  state: T;
+};
+
+interface State {
+  provider: string;
+  url: string | undefined;
+}
+
+function useGenerateAuthToken(state: State) {
+  const api = useApi();
+  const date = new Date().toISOString();
+
+  const createToken = async () =>
+    await api.requestPromise(`/api-tokens/`, {
+      method: 'POST',
+      data: {
+        name: `${state.provider} Token ${date}`,
+        scopes: [
+          'project:read',
+          'project:write',
+          'project:admin',
+          'event:read',
+          'event:write',
+          'event:admin',
+          'org:read',
+          'org:write',
+          'org:admin',
+        ],
+      },
+    });
+
+  return {createToken};
+}
+
+export function SetupIntegrationModal<T extends Data>({
+  Header,
+  Body,
+  Footer,
+  closeModal,
+}: ModalRenderProps) {
+  const [state, setState] = useState<State>({
+    provider: 'LaunchDarkly',
+    url: undefined,
+  });
+  const {organization} = useLegacyStore(OrganizationStore);
+  const {createToken} = useGenerateAuthToken(state);
+
+  const handleSubmit = useCallback(() => {
+    addSuccessMessage(t('Integration set up successfully'));
+    closeModal();
+  }, [closeModal]);
+
+  const ModalHeader = useCallback(
+    ({children: headerChildren}: {children: React.ReactNode}) => {
+      return (
+        <Header closeButton>
+          <h3>{headerChildren}</h3>
+        </Header>
+      );
+    },
+    [Header]
+  );
+
+  const ModalFooter = useCallback(() => {
+    return (
+      <Footer>
+        <StyledButtonBar gap={1}>
+          <LinkButton priority="default" href="">
```

Comment:
need to add link here cc @cmanallen - lmk if you have an expected URL in mind, or we can leave it blank for now since it's only visible to sentry anyway

Reply:
Lets use this URL:

```
https://docs.sentry.io/product/issues/issue-details/#feature-flags
```